### PR TITLE
Add Jacoco to gather coverage metrics

### PR DESCRIPTION
**Pull request by stevepeak**
Hey :) Lets see if this works

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,5 @@ sudo: required
 language: java
 jdk: oraclejdk8
 
-before_install:
-  - sudo pip install codecov
 after_success:
-  - codecov
+  - bash <(curl -s https://codecov.io/bash)
```
Comment (stevepeak) [inline .travis.yml]: I switched to the bash uploader for a little faster processing during CI. But both options work

File: pom.xml [hunk 2]
```diff
@@ -15,5 +15,30 @@
 		<module>account-service</module>
 		<module>statistics-service</module>
 	</modules>
+	
+	<build>
+		<sourceDirectory>src/main/java</sourceDirectory>
+		<plugins>
+			<plugin>
+				<groupId>org.jacoco</groupId>
```
Comment (stevepeak) [inline pom.xml]: Jacoco is required to gather metrics: https://github.com/codecov/example-java

**Comment by stevepeak** [comment]
Here is the error I've got right now:

```
[INFO] Skipping JaCoCo execution due to missing execution data file:/home/travis/build/sqshq/PiggyMetrics/target/jacoco.exec
```

https://travis-ci.org/sqshq/PiggyMetrics/builds/109089283#L3780

**Comment by stevepeak** [comment]
I'm not very experienced with Java. Here is what I found that may fix this: http://stackoverflow.com/questions/18107375/getting-skipping-jacoco-execution-due-to-missing-execution-data-file-upon-exec

**Comment by sqshq** [comment]
Hi! Thank you for your help and amazing product!

I don't use parent pom in this project because it's much nicely to use Spring Boot starter pom for each microservice. So I think there is no way to share plugin configuration between them.

I've included Jacoco plugin in each necessary pom and now everithing working good:
[codecov.io](https://codecov.io/github/sqshq/PiggyMetrics?branch=master)

**Pull request merged**